### PR TITLE
feat: simplify data pipeline — canonical types + enriched fields (#233)

### PR DESCRIPTION
**Pull request by drewdrewthis**
## Summary

Rebuilds the data pipeline per #233:
- Collapses the 4-type-per-entity mapping chain into one canonical type per entity in `src/models/`
- Expands GraphQL queries with per-branch aliases for richer PR/issue data
- Adds batched git enrichment (ahead/behind, last commit date)
- Two-file Claude telemetry merge (hook + status line)
- Decomposes derive.rs into focused modules

## Progress

- [x] Phase 1: Canonical types defined in `src/models/`
- [ ] Phase 2a: Per-branch GraphQL PR queries with aliases
- [ ] Phase 2b: Per-branch issue queries + sub-issues API
- [ ] Phase 2c: Repo-level fields + batched git enrichment
- [ ] Phase 2d: Two-file Claude telemetry merge
- [ ] Phase 3a: Decompose derive.rs
- [ ] Phase 3b: Migrate consumers
- [ ] Phase 3c: Delete old types + dead code
- [ ] Phase 3d: JSON output v3

## Test plan

- [ ] All existing tests pass
- [ ] `cargo clippy -- -D warnings` clean
- [ ] `orchard --json` output backward-compatible (v2 fields preserved)
- [ ] TUI renders correctly with new data pipeline
- [ ] Per-branch GraphQL queries return correct PR state

🤖 Generated with [Claude Code](https://claude.com/claude-code)

# Related issue

- #233

# Related issue

- #233

# Related issue

- #233

# Related issue

- #233

# Related issue

- #233

# Related issue

- #233

# Related issue

- #233

# Related issue

- #233

# Related issue

- #233

# Related issue

- #233

### CODE REVIEW

**Review by drewdrewthis — COMMENTED** [review]
## Review: simplify data pipeline — canonical types + enriched fields (#233)

**Verdict: Safe to merge. No blockers.**

CI is green (978 tests pass, clippy + fmt clean). The decomposition is clean, the enrichment pipeline is well-tested, and backward compatibility is maintained. This is solid work.

### What's good

- The derive.rs decomposition into join.rs (matching/enrichment) + classify.rs (display group logic) is a clean separation of concerns. derive.rs drops from ~1950 to ~350 lines while preserving its public API via re-exports.
- Per-branch GraphQL queries are a significant improvement over the bulk pullRequests(states:OPEN) approach — you get the most recent PR per branch regardless of state, plus repo-level metadata in one call.
- The two-file Claude telemetry merge (hook + status line) is well designed: separate channels, no locking, merged at consumption. The StatusLineFile filtering (skip .tmp files, skip non-matching filenames) is defensive.
- Test coverage is thorough — new parsers (git branch -vv, git for-each-ref, per-branch GraphQL, issue GraphQL, blocking regex, tmux timestamps) all have unit tests covering happy paths, edge cases, and error paths.
- Backward compatibility: all new fields use `#[serde(default)]` / `skip_serializing_if`, existing JSON paths are preserved, and the ListEntry enum variants are correctly boxed to avoid stack bloat from the larger WorktreeRow.

### Findings (non-blocking)

1. **models/ are defined but not consumed yet.** The canonical types in `src/models/` (Repo, Pr, Issue, Session, Worktree, Claude) coexist alongside the existing types as the PR description notes. The PR checklist shows phases 2a-3d unchecked. This is fine for an incremental approach — just noting that the 4-type-per-entity chain still exists and these models are aspirational for now. No dead code concern since they're public API surface for future phases.

2. **`parse_git_ahead_behind` returns (0, 0) for untracked branches.** A branch with no tracking remote gets `(0, 0)` which is then stored as `ahead: None, behind: None` in the worktree (because the refresh code checks `if a > 0 { Some(a) } else { None }`). This means "no tracking remote" and "in sync with remote" are indistinguishable. Acceptable for now since it degrades gracefully.

3. **`sanitize_branch_alias` collision potential.** Branches `feat/foo-bar` and `feat/foo_bar` both become `b_feat_foo_bar`. The per-branch query would silently drop one PR. This is a very unlikely real-world scenario and GraphQL would error visibly if aliases collide, so low risk.

4. **`is_default_branch` is now `pub`.** The visibility change from private to public is intentional (used by cache_sources for filtering), but the function hardcodes `main | master | develop | dev`. If a repo uses a non-standard default branch (e.g. `trunk`), those branches would get PR queries. The new `defaultBranchRef` field in the GraphQL response could be used to make this dynamic in a future pass.

5. **Dead code cleanup is clean.** session_discovery.rs and status.rs were correctly identified as orphaned and removed. No dangling references.

6. **JSON output version.** The commit message says "Bump JSON output version to 3" but I don't see a version field in the diff — the version bump may be in the orchard binary's CLI output rather than in the library. Worth confirming the `--json` output actually advertises v3 so consumers can detect the schema change.

File: crates/orchard/src/cache_sources.rs [hunk 1]
```diff
@@ -48,11 +52,178 @@ pub fn parse_issues_json(json: &str) -> Vec<CachedIssue> {
...
+  repository(owner: "{owner}", name: "{name}") {{
+{aliases}
+  }}
+}}
+
+fragment IssueFields on Issue {{
+  number
+  title
+  state
+  labels(first: 100) {{ nodes {{ name }} }}
```
Comment (drewdrewthis) [inline crates/orchard/src/cache_sources.rs:94]: Branches `feat/foo-bar` and `feat/foo_bar` both produce alias `b_feat_foo_bar`. GraphQL would reject duplicate aliases. Low risk but worth a doc comment.

File: crates/orchard/src/cache_sources.rs [hunk 2]
```diff
@@ -354,6 +552,8 @@ pub fn parse_tmux_output(
             window_names: parsed.window_names,
             window_active: parsed.window_active,
             host: host.map(|h| h.to_string()),
+            created_at,
+            last_activity_at,
             last_output_lines,
```
Comment (drewdrewthis) [inline crates/orchard/src/cache_sources.rs:557]: Branches with no tracking remote and branches perfectly in-sync both produce `(0, 0)`. Consider returning `None` for the whole entry when no bracket section is found, to distinguish "no remote" from "in sync with remote".
